keel-core/src/main/kotlin/com/netflix/spinnaker/keel/lifecycle/LifecycleMonitor.kt

```diff
@@ -61,4 +48,40 @@ abstract class LifecycleMonitor(
     log.debug("${this.javaClass.simpleName} has completed monitoring for $task")
     monitorRepository.delete(task)
   }
+
+  /**
+   * Handles persisting failures and publishing exception events.
+   *
+   * Call when there was a failure getting status.
+   */
+  fun handleFailureFetching(task: MonitoredTask) {
+    if (task.numFailures >= lifecycleConfig.numFailuresAllowed - 1) {
```

Comment:
I know the original logic was already like this, but it strikes me as odd that we're comparing with `>=` and _also_ subtracting `1`. Let's say `lifecycleConfig.numFailuresAllowed` is 3. We'll consider it an error if `task.numFailures` is equal to or more than 2. That doesn't sound right, unless we're doing something weird when counting failures...

Reply:
That does seem pretty confusing. Isn't it the same as `if (task.numFailures > lifecycleConfig.numFailuresAllowed)`?

Reply:
So this is called when we have a failure, which means the number of failures that we have is actually one greater than what is saved on the task. If we only want to allow 3 failures, we shouldn't try a fourth time (I think). So this was how I wrote the logic to do that.

In the case that you wrote @robfletcher, we'd allow one more failure than the config states

Reply:
I can't math

Reply:
I think that just means you should be incrementing `task.numFailures` before you reach this line...